lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -34,9 +36,10 @@
 from sycamore.utils.memory_debugging import display_top, gc_tensor_dump
 from sycamore.utils.pdf import convert_from_path_streamed_batched
 from sycamore.utils.time_trace import LogTime, timetrace
-
+from sycamore.utils.pytorch_dir import get_pytorch_build_directory
 
 logger = logging.getLogger(__name__)
+_DETR_LOCK_FILE = "~/.cache/Aryn-Detr.lock"
```

Comment:
Beware that not everything interprets tilde (~) as $HOME.  It's safer to use os.environ["HOME"] or to pick a fixed path like in /tmp.  If you really want to do this right, use pwd.getpwuid(os.getuid()).pw_dir.  I'd convert this string constant to a suffix and put the home-dir logic below.

Reply:
Done.